pkg/deploy/api/validation/validation.go

```diff
@@ -37,7 +37,7 @@ func ValidateDeploymentConfig(config *deployapi.DeploymentConfig) fielderrors.Va
 	if config.Spec.Replicas < 0 {
 		allErrs = append(allErrs, fielderrors.NewFieldInvalid("spec.replicas", config.Spec.Replicas, "replicas cannot be negative"))
 	}
-	if config.Spec.Selector == nil || len(config.Spec.Selector) == 0 {
+	if len(config.Spec.Selector) == 0 {
```

Comment:
If selector == nil and template.labels == nil then after defaulting->converting selector==nil and this line will invalidate the dc.